torchvision/csrc/io/image/cuda/decode_jpegs_cuda.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include <torch/types.h>
+#include <vector>
+#include "../image_read_mode.h"
+
+#if NVJPEG_FOUND
+#include <c10/cuda/CUDAStream.h>
+#include <nvjpeg.h>
+
+namespace vision {
+namespace image {
+class CUDAJpegDecoder {
+ public:
+  CUDAJpegDecoder(const torch::Device& target_device);
+  ~CUDAJpegDecoder();
+
+  std::vector<torch::Tensor> decode_images(
+      const std::vector<torch::Tensor>& encoded_images,
+      const nvjpegOutputFormat_t& output_format);
+
+  const torch::Device original_device;
+  const torch::Device target_device;
+  const c10::cuda::CUDAStream stream;
+
+ protected:
```

Comment:
Out of curiosity, why protected and not private?

Reply:
No real reason. Happy to change to private